pkg/server/plugin/nodeattestor/aws/iid.go

```diff
@@ -385,3 +395,108 @@ func unmarshalAndValidateIdentityDocument(data []byte, pubKey *rsa.PublicKey) (e
 
 	return doc, nil
 }
+
+func (p *IIDAttestorPlugin) ResolveSelectors(ctx context.Context, region, instanceID string) (*common.Selectors, error) {
+	client, err := p.clients.getClient(region)
+	if err != nil {
+		return nil, iidError.New("failed to get client: %w", err)
+	}
+
+	resp, err := client.DescribeInstancesWithContext(ctx, &ec2.DescribeInstancesInput{
```

Comment:
This call may have already happened during the Attest call (albeit with different filters). I think we should refactor the code to always call DescribeInstancesWithContext during the Attest call (instead of conditionally), using these filters, since we're going to be need it to build the selector list anyway.